LambdaRuntimeDockerfiles/Infrastructure/src/Infrastructure/PipelinesStage.cs

```diff
@@ -13,13 +13,42 @@ public PipelinesStage(
         string id,
         Configuration configuration,
         IStageProps props = null) : base(scope, id, props)
+    {
+        // Create pipelines for main repository
+        CreatePipelinesForRepository(configuration, 
+            configuration.ProjectName,
+            configuration.GitHubOwner, 
+            configuration.GitHubRepository, 
+            configuration.GitHubBranch);
+        
+        // Create pipelines for staging repository
+        CreatePipelinesForRepository(configuration,
+            $"{configuration.ProjectName}-staging",
+            configuration.GitHubOwnerStaging,
+            configuration.GitHubRepositoryStaging,
+            configuration.GitHubBranchStaging);
+    }
+
+    private void CreatePipelinesForRepository(
+        Configuration configuration, 
+        string pipelinePrefix,
+        string gitHubOwner, 
+        string gitHubRepository, 
+        string gitHubBranch)
     {
         for (var i = 0; i < configuration.Frameworks.Length; i++)
         {
+
+            var pipelineName = $"{pipelinePrefix}-{configuration.Frameworks[i].Framework}";
+
             new PipelineStack(this,
-                configuration.Frameworks[i].Framework,
+                $"{pipelinePrefix}-{configuration.Frameworks[i].Framework}",

```

Comment:
cloudformation automatically uses the parent stack + this for the name of the stack and we get something like 

```
aws-lambda-container-images-aws-lambda-container-images-staging-net9
```

but i dont really care about the stack name too much to be honest. This was the only way i found to make the pipeline name + what appears in the ui for the pipeline in the main pipeline look correct